main/main.cpp

```diff
@@ -92,7 +98,29 @@ int main(int argc, char** argv) {
             auto filename = evaluate_command.get<std::string>("file");
             LOG_D() << "file present: " << filename;
             BinFileRead bfr{filename};
-            bfr.run<uint32_t>();
+            if (evaluate_command.get<std::string>("type").find("u32") !=
+                std::string::npos) {
+                bfr.run<uint32_t>();
+                return 0;
+            }
+
+            if (evaluate_command.get<std::string>("type").find("int32") !=
+                std::string::npos) {
+                bfr.run<int32_t>();
+                return 0;
+            }
+
+            if (evaluate_command.get<std::string>("type").find("f32") !=
+                std::string::npos) {
+                bfr.run<float>();
+                return 0;
+            }
+
+            if (evaluate_command.get<std::string>("type").find("b") !=
+                std::string::npos) {
+                bfr.run<bool>();
+                return 0;
+            }
```

Comment:
type names do not follow a consistent pattern (u32, int32?)

also no error handling for unkown pattern